### DISCUSSION TITLE
Warn against opcache stripping comments in System Report and/or docs.

### DISCUSSION THREAD

**Discussion by mmikkel** [craftcms/cms#13631] [Ideas] [answered]
In cases where OPcache is enabled and [configured to not save comments](https://www.php.net/manual/en/opcache.configuration.php#ini.opcache.save-comments), **Craft isn't able to render markdown; i.e. it will be output verbatim instead of being rendered as HTML.**

The reason for that is that the `cebe/markdown` package, which [mentions this caveat in its readme file](https://github.com/cebe/markdown#installation-), "uses PHPDoc annotations to determine the markdown elements that should be parsed".

In practice, this means that (for example) the email links in Craft's system messages will not be output as actual `<a>` tags, but like this: `<https://example.com/admin/set-password?code=...>` – which makes most email clients simply neglect to display them, and content authors despairing.

The fix is simple: if using OPcache, PHP needs to be configured with `opcache.save_comments=1`. But this is a bit of a gotcha that might not be immediately obvious (wasn't to me, in any case 😅).

**Would it make sense to add a warning to the System Report in the Control Panel in cases where OPcache is enabled and configured to strip comments?** If not, or additionally, make a note of it somewhere in the Craft docs?

**Comment by brandonkelly** [comment, marked as answer]
Thanks for pointing that out!

Just tagged a new version of `craftcms/server-check` with an added OPcache + `opcache.save_comments` requirement (https://github.com/craftcms/server-check/commit/445c8eccde570ede09dc96c39fc1357a5296520f).

**Comment by vardumper** [comment]
@samdark Would it be possible to get rid of this package and use Twigs' twig/markdown-extra instead? `{% apply markdown_to_html instead %}`
